### ISSUE TITLE
Tooltip is clipped when the anchor widget is near the edge of the window

### ISSUE THREAD

**Issue by golovinengineer** [slint-ui/slint#12260]
### Bug Description

The new Tooltip element (introduced in 1.17) positions itself to the right of the mouse cursor by default. When the widget it's attached to is near the right edge of the window, the tooltip overflows past the window boundary and gets visually cut off instead of being repositioned to stay fully on-screen or going behind window's edge.

## Steps to reproduce:
```slint
import { Button } from "std-widgets.slint";

export component MainWindow inherits Window {
    width: 300px;
    height: 200px;

    Button {
        text: "Save";
        x: parent.width - self.width - 4px;
        y: 50px;
        Tooltip {
            text: @markdown("Save the current document (Ctrl+S)");
        }
    }
}
```

Run the app or run with slint-viewer.
Hover over the button near the right edge.
The tooltip opens to the right of the cursor and is clipped by the window edge.

Worth noting that I cannot recreate in SlintPad at https://slintpad.com

## Expected behavior:
The tooltip should detect it would overflow the window bounds on the right and flip to open on the left side of the cursor instead (similar to how native OS tooltips/popups reposition themselves near screen edges).
Actual behavior:

The tooltip is rendered partially outside the window and is clipped.

### Reproducible Code (if applicable)

```slint
import { Button } from "std-widgets.slint";

export component MainWindow inherits Window {
    width: 300px;
    height: 200px;

    Button {
        text: "Save";
        x: parent.width - self.width - 4px;
        y: 50px;
        Tooltip {
            text: @markdown("Save the current document (Ctrl+S)");
        }
    }
}
```

### Environment Details

- Slint Version: 1.17.0
- Platform/OS: macOS
- Programming Language: Python
- Backend/Renderer: Winit + Skia

### Product Impact

Non-blocker, but break the look

**Comment by Murmele** [comment]
Thanks for reporting this issue. This will be fixed in https://github.com/slint-ui/slint/pull/11740

**Comment by golovinengineer** [comment]
Cool!
